### PR TITLE
Dev spend choc

### PR DESCRIPTION
**Pull request by john-baxter**

### CODE REVIEW

File: player.py [hunk 1]
```diff
@@ -576,3 +571,330 @@ def score_wine(self, standard_tokens_for_trail, wine_rule):
...
+      for i in range(2):
+        allowed_choices_direction = self.define_allowed_choices_direction(ant, trail, ant_positions)
+        direction = self.make_choice(allowed_choices_direction, PROMPT_TEXT_GAME_CHOICE_DIRECTION)
+        (food_to_hand, trail) = self.take_food_from_trail(trail, ant_positions, ant, direction)
+        self.store_food(food_to_hand)
+
+    return (trail, ant_positions, anthill, anthill_food_tokens)
+
+  def go_to_anthill(self, trail, ant_positions, anthill, anthill_rule, ant, anthill_food_tokens):
+    """Performs actions foran ant to move onto the anthill.
```
Comment (anaandresarroyo) [inline player.py:825]:
```suggestion
    """Performs actions for an ant to move onto the anthill.
```

File: player_test.py [hunk 2]
```diff
@@ -1262,5 +1263,708 @@ def test_score_wine_calls_oenophile_method_from_constants(self):
...
+    mock_ask.assert_called_once_with()
+
+  @patch('player.Player.ask_to_spend_chocolate')
+  def test_will_spend_choc_only_calls_ask_to_spend_if_choc_in_hand(self, mock_ask):
+    # test 212
+    mario = Player("Mario")
+    mario.hand = {"cheese": 3}
+
+    mario.will_spend_choc()
+    assert not mock_ask.called
```
Comment (anaandresarroyo) [inline player_test.py:1398]:
```suggestion
    mock_ask.assert_not_called()
```

File: player_test.py [hunk 3]
```diff
@@ -1262,5 +1263,708 @@ def test_score_wine_calls_oenophile_method_from_constants(self):
...
+    ant = "yellow"
+    ant_positions = {"yellow": 0}
+    anthill = [None]
+    anthill_rule = ""
+    anthill_food_tokens = {"cheese" : 1}
+    mario = Player("Mario")
+
+    manager = mock.Mock()
+    manager.attach_mock(mock_move_ant, 'mock_move_ant')
+    manager.attach_mock(mock_move_ant, 'mock_move_ant')
```
Comment (anaandresarroyo) [inline player_test.py:1458]: Attach each mock only once.
```suggestion
```

File: player_test.py [hunk 4]
```diff
@@ -1262,5 +1263,708 @@ def test_score_wine_calls_oenophile_method_from_constants(self):
...
+    trail = ["cheese", "apple", "bread"]
+    ant = "yellow"
+    ant_positions = {"yellow": 0}
+    anthill = [None]
+    anthill_rule = ""
+    anthill_food_tokens = {"cheese" : 1}
+    mario = Player("Mario")
+
+    manager = mock.Mock()
+    manager.attach_mock(mock_allowed_ants, 'mock_allowed_ants')
```
Comment (anaandresarroyo) [inline player_test.py:1505]: I can see why you have used these mocks if you wrote this test before refactoring sections into other methods such as `choose_ant()` or `go_to_anthill()`. Consider refactoring this test (and other similar tests) to mock the methods called by `take_turbo_turn()` directly only (e.g. `choose_ant()`), not those in deeper levels (e.g. `make_choice()`).

File: player_test.py [hunk 5]
```diff
@@ -1262,5 +1263,708 @@ def test_score_wine_calls_oenophile_method_from_constants(self):
...
+    manager.attach_mock(mock_make_choice, 'mock_make_choice')
```
Comment (anaandresarroyo) [inline player_test.py:1910]: Attach eat mock only once.

File: controller_test.py [hunk 6]
```diff
@@ -104,41 +104,64 @@ def test_for_3_players_generate_player_is_called_three_times(
...
+    mock_prepare_players,
+    mock_choose_game_rule,
+    mock_bites_init,
+    mock_bites_play,
+    ):
+    # test 231
+
+    manager = mock.Mock()
+    manager.attach_mock(mock_prepare_players, 'mock_prepare_players')
+    manager.attach_mock(mock_choose_game_rule, 'mock_choose_game_rule')
```
Comment (anaandresarroyo) [inline controller_test.py:138]: Attach each mock only once.

File: bites_test.py [hunk 7]
```diff
@@ -343,110 +438,117 @@ def test_first_player_takes_one_turn(self, render_game_mock):
...
-      "grapes"]
-    starting_ant_positions = {
-      "red": None,
-      "yellow": None,
-      "green": None,
-      "brown": None,
-      "purple": None}
-    starting_anthill = [None, None, None, None, None]
-    starting_anthill_food_tokens = {}
+  # @patch('bites.Bites.render_game')
```
Comment (anaandresarroyo) [inline bites_test.py:441]: Remove obsolete / commented code.

**Review by anaandresarroyo — APPROVED** [review]

File: player_test.py (hunk 4 above)
Comment (john-baxter) [reply player_test.py:1505]: I chose to leave this as it was.

**Pull request merged**